### PR TITLE
feat(theming)!: add `variables` to theme `colors`

### PR DESCRIPTION
**Pull request by jzempel**
## Description

This PR adds a new `variables` section to Garden's `colors` theme. These variables will provide the precedent lookup for most Garden component colors. The BREAKING CHANGE is the removal of `colors.background` and `colors.foreground` – now obviated by the v9 variables.

## Detail

The forthcoming `getColor` utility replacement PR will utilize the new `variables` to obtain light/dark mode-based colors.

## Checklist

- [ ] :ok_hand: ~design updates will be Garden Designer approved (add the designer as a reviewer)~
- [ ] :globe_with_meridians: ~demo is up-to-date (`npm start`)~
- [ ] :arrow_left: ~renders as expected with reversed (RTL) direction~
- [ ] :metal: ~renders as expected with Bedrock CSS (`?bedrock`)~
- [x] :guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)
- [ ] :wheelchair: ~tested for WCAG 2.1 AA accessibility compliance~
- [ ] :memo: ~tested in Chrome, Firefox, Safari, and Edge~

### CODE REVIEW

File: packages/colorpickers/src/styled/ColorSwatch/StyledColorSwatchLabel.ts
Comment (jzempel) [inline packages/colorpickers/src/styled/ColorSwatch/StyledColorSwatchLabel.ts:1]: FYI this file is new on the `next` branch and did not receive the `getColorV8('background/foreground')` treatment with #1756 on `main`.

File: packages/theming/src/types/index.ts [hunk 1]
```diff
@@ -75,14 +75,24 @@ export interface IGardenTheme {
...
+        background: Record<string, string>;
+        border: Record<string, string>;
+        foreground: Record<string, string>;
+      };
+      light: {
+        background: Record<string, string>;
+        border: Record<string, string>;
+        foreground: Record<string, string>;
+      };
+    };
```
Comment (geotrev) [inline packages/theming/src/types/index.ts:95]: How are you feeling about the `variables` property name? I was thinking about the [CSS naming](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) the other day (e.g., `scheme` from `prefers-color-scheme`). I don't feel particularly strongly, just curious if that's come up for you at all.

Comment (geotrev) [reply packages/theming/src/types/index.ts:95]: Also thinking about how this writes in our components:

- `p.theme.colors.variables.light.foreground`
- `p.theme.colors.scheme.light.foreground`
- etc

😅

Comment (jzempel) [reply packages/theming/src/types/index.ts:95]: > How are you feeling about the `variables` property name? I was thinking about [CSS naming](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) the other day (e.g., `scheme` from `prefers-color-scheme`). I don't feel particularly strongly, just curious if that's come up for you at all.

I don't like `scheme` as this object is not a scheme – it's a collection of variables. We're leaning into the design language here and making room for additional themes such as "lightHighContrast", "darkDimmed", etc.

Comment (jzempel) [reply packages/theming/src/types/index.ts:95]: > Also thinking about how this writes in our components:
> 
> * `p.theme.colors.variables.light.foreground`
> * `p.theme.colors.scheme.light.foreground`
> * etc
> 
> 😅

We will never reference values in this manner. It should make more sense when the `getColor` PR lands, but the function call will be something like `getColor({ theme, variable: 'foreground.default' });`

Comment (geotrev) [reply packages/theming/src/types/index.ts:95]: > I don't like scheme as this object is not a scheme

Makes sense, especially with high contrast and other themes in the variable list. 👌🏻

**Review by geotrev — APPROVED** [review]

**Pull request merged**